OasisDEX/oasis-borrow
# Lw/6664 follow api

# [Follow API](https://app.shortcut.com/oazo-apps/story/6664/create-api-endpoints-for-followed-vaults-following)
# [Follow button handling](https://app.shortcut.com/oazo-apps/story/7143/follow-vault-using-ui-api)
  
## Changes 👷‍♀️
- GET endpoint to fetch followed vaults
- POST endpoint to Follow vault
- Follow/Following/Unfollow button in VaultHeadline
- Managing FollowButton state depending if vault is being followed, processing or not followed, unfollow when vault is followed and mouse is hovering on button
  
## How to test 🧪
- FollowVaults flag must be turned on
![image](https://user-images.githubusercontent.com/6871923/208969485-28b11c3a-43ef-46bf-81f0-e8850462b164.png)

- Go to vault page of vault you want to follow, NOTE vault must exist in db otherwise endpoint will return 500, as not being able to create follow record for non-existing vault
![image](https://user-images.githubusercontent.com/6871923/208970017-673e2ecf-817d-4c94-ba16-167c279bdd18.png)

- Note that when opening vault page, button should be in Loading state when it's checking if vault is followed or not
- Open a vault that is not yet followed by your account (there's no record with this vault in `users_who_follow_vaults` table)
![image](https://user-images.githubusercontent.com/6871923/208970408-04de8c9e-31c9-46c9-89b0-7a75ef2fe342.png)

- Click follow, check if button is in loading state when waiting for response from server
![image](https://user-images.githubusercontent.com/6871923/208970550-21664037-1d3f-4bd5-8185-b295f32cb37c.png)

- After following check if record with vault of your preference is added to db and button is in `Following` state
![image](https://user-images.githubusercontent.com/6871923/208970710-ac3fef4d-ae67-46bc-8618-37efcf7ac4f3.png)

- Check if button is switching state to `Unfollow` upon hovering on it with mouse
![image](https://user-images.githubusercontent.com/6871923/208970831-2dc1a1f0-4049-407f-927c-66a7f270cc44.png)
